First version of a pickup magnet

Kevin22888 opened the pull request:
- A point effector 2D and a circle collider, with some parameter tweaking.
- Fixed a small bug where pickups stick to the walls (due to a function that is reducing movement, which is now removed)

Interesting Note:
When a pickup is starting to get attracted towards the player, if we quickly run/jump away, the pickup will be left behind.
This may not necessary make the magnet that effective, so we could consider some more changes.

eon704 commented:
Magnet feels nice!

A little comment:
- At times it feels that the magnet's range needs to be longer - how does it feel if we increase the magnet range by 50%?

Kevin22888 commented:
Changes:
- Removed Point Effector and extra collider and added a custom script for the magnet effect

`ResourceMagnet` script on the player:
- can customize the attraction radius, and min/max speed
- use Lerp to set attraction speed based on min/max settings

Eryngii01 commented on Assets/Code/Scripts/Core/Player/ResourceMagnet.cs:
```diff
@@ -0,0 +1,74 @@
+                    float distanceToPlayer = directionToPlayer.magnitude;
+                    
+                    // Choose a speed based on the pickup's distance to the player
+                    float speed = Mathf.Lerp(_minSpeed, _maxSpeed, distanceToPlayer / _attractionRadius);
```
I believe that when the resource pickup is farthest from the player's attraction radius, the `distanceToPlayer` will be maximized and will begin moving at the `_maxSpeed` which will eventually lerp to the `_minSpeed` as the pickup moves closer. 

If this was intended, feel free to disregard this comment 👍

eon704 replied:
Personally, it appears visually better to my eye as it is, starting at the max speed and then gradually slowing down 🤔

eon704 reviewed (approved):
Overall looks good!

Personally, I think the value of 40 for the `MaxSpeed` is a bit too much. I would put it somewhere in the range of 20-25. And as for the `MinSpeed` I think having it around 12-14 appears quite nice. But these are personal preferences, if you would like to change those to something else feel free to do so 👌
